app/socket-apis/get-user-whys.js

```diff
@@ -0,0 +1,39 @@
+// https://github.com/EnCiv/civil-pursuit/issues/206
+
+const Points = require('../models/points')
+async function getUserWhys(ids, cb) {
+  const cbFailure = errorMsg => {
+    if (errorMsg) console.error(errorMsg)
+    if (cb) cb(undefined)
+  }
+
+  // Check if user is logged in
+  if (!this.synuser || !this.synuser.id) {
+    return cbFailure('Cannot retrieve whys - user is not logged in.')
+  }
+
+  // Verify that ids is provided and is a valid array
+  if (!Array.isArray(ids) || ids.length === 0) {
+    return cbFailure('Invalid argument provided to getUserWhys(ids: Array, cb: Function).')
+  }
+
+  // Fetch points from the Points collection where parentId matches any of the ids
+  let pointsList
+  try {
+    pointsList = await Points.find({ parentId: { $in: ids }, userId: this.synuser.id }).toArray()
+  } catch (error) {
+    return cbFailure('Failed to retrieve points - Points.find failed.')
+  }
```

Comment:
This try / catch block is correct.  But I want to ask you to try something else that would be cleaner .. if it works:
```
const pointsList = await Points.find(...).toArray().catch(error=>cbFailure(`Failed to retrieve points - Points.find failed: {$error}`)
```
Also notice that in the example it's added the text of the error message into the message that's going to cbFailure. This can be done in other places too.